storybook/qmlTests/TestComponents/tst_StatusChatInput.qml

```diff
@@ -1,21 +1,571 @@
+        }
+    }
+
+    TestCase {
+        id: statusChatInputKeyboardInputExpectedAsText
+        name: "StatusChatInputKeyboardInputExpectedAsText"
+        when: windowShown
+
+        property StatusChatInput controlUnderTest: null
+
+        function init() {
+            Utils.globalUtilsInst = globalUtilsMock
+            controlUnderTest = createTemporaryObject(componentUnderTest, root)
+        }
+
+        //Scenario: StatusChatInput will display any typed Ascii visible character
+        //Given a new instance of StatusChatInput
+        //When the user is typing any Ascii visible character as <text>
+        //Then the text is displayed as it is typed
+        //And the input text is not modified by mentions processor
+        //And the input text is not modified by emoji processor
+        //
+        //Example:
+        //text
+        //(!"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\]^_`abcdefghijklmnopqrstuvwxyz{|}~)
+        function test_keyboard_any_ascii_input() {
+            var expectations_after_typed_text = (typedText) => {
+                verify(controlUnderTest.getPlainText() === typedText,
+                                           `Expected text: ${typedText}, received: ${controlUnderTest.getPlainText()}`)
+                verify(controlUnderTest.textInput.text === controlUnderTest.getTextWithPublicKeys(),
+                                            `Expected text: ${controlUnderTest.textInput.text}, received: ${controlUnderTest.getTextWithPublicKeys()}`)
+            }
+
+            testHelper.when_text_is_typed(statusChatInputKeyboardInputExpectedAsText,
+                                          testHelper.get_all_ascii_characters(),
+                                          expectations_after_typed_text);
+        }
+
+//        Scenario outline: User can navigate in chat input using keyboard
+//        Given the user has typed text on multiple lines in StatusChatInput
+//        """
+//        Typing on first row
+//        Typing on the second row
+//        Typing on the third row
+//
+//        """
+//        When the user hits <direction>
+//        The cursor will move inside the text area to the <direction>
+//
+//        Examples:
+//        |direction|
+//        |left|
+//        |right|
+//        |up|
+//        |down|
+
+//        When the user hits "left"
+//        The cursor will move inside the text area to the "left"
+//        When the user hits "right"
+//        The cursor will move inside the text area to the "right"
+//        When the user hits "up"
+//        The cursor will move "up" inside the text area, on the previous line if any
+//        When the user hits "down"
+//        The cursor will move "down" inside the text area, on the next line if any
+        function test_user_can_navigate_with_keys() {
+
+            const firstRowText = "Typing on first row"
+            testHelper.when_text_is_typed(statusChatInputKeyboardInputExpectedAsText,
```

Comment:
Could you here use instead `when_multiline_text_is_typed`? 
As a general comment, could it make sense to create a global object, where data that is used in more than one scenario, is defined, so you can just call the needed one in each scenario? I mean something like this:

````
    QtObject {
        id: testData
        
        readonly property string firstRowText: "Typing on first row"
        readonly property string secondRowText: "Typing on the second row"
        readonly property string thirdRowText: "Typing on the third row"
        //...        
    }

````

Reply:
Using a general data object would definitely improve reusability and probably slightly improve the run time. But I think it would affect a bit the code readability and the tests will not be independent of each-other.

I have tried to keep the test data closer to the expectation in order for the test to be easily read. The goal was to be able to read the test implementation as you read the scenario outline and I've also tried to keep the implementation as simple as possible and to use hardcoded expectations. One other consideration I had was to keep tests independent of each other. So the result of all this was to define data in the scenario outline and hardcode the data in the test (or in the data object when the same test runs on multiple data sets).

Reply:
Ok. It was just a suggestion, feel free to keep it or discard!

However, I don't see an issue to create this generic data object and use it in the scenarios you want. You don't lose the indpenendence between test cases, if you need specific that for one of them, then just write the specific data for it. At the end you are writting the same const properties a lot of times when you could just have them in one place! 
I understand your point related to readability but I neither see an issue with that by using the global data!

Decision totally up to you! :)

Reply:
Cool, thanks! Looks like you are right and it looks good.
Updated based on your suggestions. Please take a look.
I also added a given_multiline_text_is_typed as most test cases will not set expectation while the text is typed.